pkg/platform/runtime/setup/implementations/alternative/artifact.go

```diff
@@ -23,7 +23,23 @@ func NewArtifactSetup(artifactID artifact.ArtifactID, store *store.Store) *Artif
 	return &ArtifactSetup{artifactID, store}
 }
 
-func (as *ArtifactSetup) EnvDef(tmpDir string) (*envdef.EnvironmentDefinition, error) {
+func (as *ArtifactSetup) PrepareEnvDef(tmpDir, installDir string, constants envdef.Constants) (*envdef.EnvironmentDefinition, error) {
+	// Source the environment definition from the extracted archive
+	ed, err := as.newEnvDef(tmpDir)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not load environment definitions for artifact.")
+	}
+
+	// Expand the environment definition variables
+	ed = ed.ExpandVariables(constants)
```

Comment:
Why was this moved into the envdef function but ApplyFileTransformations wasn't?

Reply:
This function operates on the environment definition itself whereas ApplyFileTransforms operates on the files passed to it and uses the environment definition. The idea with the prepare function was to do all the preparatory steps required on the environment definition so that it's ready to be used with artifact installation.